### PR TITLE
Fix AdamW import error for <1.2

### LINKED ISSUE #1585: AdamW requires torch>=1.2.0
## 🐛 Bug

AdamW requires torch>=1.2.0, torch < 1.2.0 will cause an importError: cannot import name 'AdamW'
Model I am using (Bert, XLNet....):

Language I am using the model on (English, Chinese....):

The problem arise when using:
* [ ] the official example scripts: (give details)
* [ ] my own modified scripts: (give details)

The tasks I am working on is:
* [x] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details)

## To Reproduce

1.
2.
3.

## Environment

* OS:
* Python version:
* PyTorch version:
* PyTorch Transformers version (or branch):
* Using GPU ?
* Distributed of parallel setup ?
* Any other relevant information:

### PR DESCRIPTION
**Pull request by BramVanroy**
closes #1585

### CODE REVIEW

**Comment by BramVanroy** [comment]
I just realized that it's better to try to import AdamW in optimization, and if not available define the custom AdamW class.

File: transformers/optimization.py [hunk 1]
```diff
@@ -103,87 +103,90 @@ def lr_lambda(self, step):
...
-                #
-                # Instead we want to decay the weights in a manner that doesn't interact
-                # with the m/v parameters. This is equivalent to adding the square
-                # of the weights to the loss with plain (non-momentum) SGD.
-                # Add weight decay at the end (fixed version)
-                if group['weight_decay'] > 0.0:
-                    p.data.add_(-group['lr'] * group['weight_decay'], p.data)
-
-        return loss
+        def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-6, weight_decay=0.0, correct_bias=True):
```
Comment (thomwolf) [inline transformers/optimization.py]: Are we sure they both have the same signature (haven't checked the new AdamW of PyTorch yet)?

Comment (BramVanroy) [reply transformers/optimization.py]: Good call; I hadn't checked. In torch's implementation, the default values differ for these values:

- `eps`: 1e-08
- `weight_decay`: 0.01

Just like for the Adam optimizer, they also provide the `amsgrad` flag.

How would you like to proceed with this?

Comment (thomwolf) [reply transformers/optimization.py]: As long as the argument names are identical, I think we could have different default values (and not have `amsgrad` is ok IMO).

**Pull request closed without merging**